src/python/pants/engine/platform.py

```diff
@@ -18,7 +20,7 @@ def current(cls) -> 'Platform':
 
   @memoized_property
   def runtime_lib_path_env_var(self):
-    return self.match({
+    return match(self, {
```

Comment:
I guess, but it seems much more straight forward to say:
```python
class Platform(Enum):
  darwin = ("darwin", "DYLD_LIBRARY_PATH")
  linux = ("linux", "LD_LIBRARY_PATH")

  def __new__(cls, value, runtime_lib_path_env_var) -> 'Platform':
    instance = super().__new__(cls, value)
    instance._runtime_lib_path_env_var = runtime_lib_path_env_var
    return instance

  @property
  def runtime_lib_path_env_var(self):
    return self._runtime_lib_path_env_var
```

Or even:
```python
@dataclass(frozen=True)
class PlatformInfo:
  name: str
  runtime_lib_path_env_var: str


class Platform(Enum):
  darwin = PlatformInfo("darwin", "DYLD_LIBRARY_PATH")
  linux = PlatformInfo("linux", "LD_LIBRARY_PATH")
  
  # Optional for less code ripple, but arguable kill and have users say:
  # `platform_instance.value.runtime_lib_path_env_var` themselves.
  @property
  def runtime_lib_path_env_var(self):
    return self.value.runtime_lib_path_env_var
```

IOW if the enum value passed to match is `self`, that seems like a smell that the enum value should be a struct instead of a single value. If you agree I'd definitely follow up and not scope creep this PR.

Reply:
I agree and will move to a followup PR!